demos-europe/demosplan-core
# F t30460 automatically remove unauthorized case worker

**Ticket:** https://yaits.demos-deutschland.de/T30460

This is the second attempt to get this fixed. The first attempt was never merged and had some weird edits going on, so @martendemos and me decided to just redo it and maybe improve a bit on the original design (original #436).

It is possible that a user was assigned as caseworker for a number of statements. Later that user gets removed from the list of authorized users for the procedure. At this point, the user should also no longer show up as caseworker on any statements, so we have to find those statements and remove the caseworker there. 

That's what this does.

### How to review/test
1. Add a user to a procedure
2. Assign that user as caseworker to a statement
3. Remove the user from the list of authorized user for that procedure (in procedure settings)
4. The user should also be removed as the caseworker on the statement

### Linked PRs (optional)
- First Draft #436

### Tasks (optional)


### PR Checklist


Delete the checkbox if it doesn't apply/isn't necessary.
